### PR TITLE
add decline code to stripe api Card error

### PR DESCRIPTION
**Pull request by thecrogers**
This PR will add direct access to decline_code from Error\Card. To give a better understanding of Card Errors.

### CODE REVIEW

File: lib/Error/Card.php [hunk 1]
```diff
@@ -27,4 +28,9 @@ public function getStripeParam()
     {
         return $this->stripeParam;
     }
+
+    public function getStripeDeclineCode()
+    {
+        return $this->stripeDeclineCode;
```
Comment (GrahamCampbell) [inline lib/Error/Card.php]: This property is not defined?

Comment (thecrogers) [inline lib/Error/Card.php]: @GrahamCampbell are you asking a question or making a statement.

Comment (GrahamCampbell) [inline lib/Error/Card.php]: I'm saying I think you forgot to define it.

Comment (thecrogers) [inline lib/Error/Card.php]: @GrahamCampbell thanks for pointing that out, but I am following the style that is used in the other exception classes. Where the property requestId is not defined either. See this file for reference https://github.com/stripe/stripe-php/blob/master/lib/Error/Base.php#L21

**Comment by brandur** [comment]
@thecrogers Thanks for the pull!

Looks mostly good to me, but I'm not too crazy about the inconsistency of pulling `decline_code` out of `jsonBody` in the card error's constructor when other special fields are extracted in `handleApiError` and then injected in. What do you think about changing that around?

**Comment by thecrogers** [comment]
@brandur Thanks for the quick response. 

That is what I had thought initially as well, but was unsure if changing the constructor was the best idea. If you are ok with changing the constructor for Card Error, I am good with that as well.

**Comment by brandur** [comment]
Hey @thecrogers, actually hmm, that's a good point. I was thinking/hoping that our errors might be an "internal enough" interface that we could change it, but there's probably a good number of users who are initializing them in tests and the like.

I'm going to pull this in as is and add a comment that we should probably change it on the next major version bump.

**Comment by brandur** [comment]
Actually, I added one more tweak in #281 to rename `StripeDeclineCode` to `DeclineCode`. I can see that an effort was made to follow the naming conventions of the other parameters in this class, but I think that those were given a "Stripe" prefix to disambiguate them (i.e. a Stripe "code" versus an HTTP "code"). Because a decline code is not ambiguous, it's safe to strip the prefix.

**Comment by brandur** [comment]
Released as 3.20.0.

**Pull request merged**